src/components/Filters/DimensionFilter.tsx

```diff
@@ -25,15 +27,91 @@ const DimensionFilter: FC<DimensionFilterProps> = ({
 
   const toggleDimension = (dimension: string) => {
     const newSelectedKeys = selectedKeys.includes(dimension)
-      ? selectedKeys.filter(key => key !== dimension)
-      : [...selectedKeys, dimension];
+    ? selectedKeys.filter(key => key !== dimension)
+    : [...selectedKeys, dimension];
     setSelectedKeys(newSelectedKeys);
     dispatch({
       type: "SET_DIMENSIONS",
       property,
       dimensions: newSelectedKeys,
     });
   };
+  
+  const handleSelectionChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newMultiSelect: string[] = e.target.value.split(",")
+    setSelectedKeys(newMultiSelect);
+    // If the rco filter includes more than the rco name, 'newDimensions' should be passed to dispatch's 'dimensions' instead of newMultiSelect.  Also in that case, 'newDimensions' should also be added to 'toggleDimension' to filter out rco dimensions.
+    // const newDimensions = (property === 'neighborhood') ? newMultiSelect : newMultiSelect.map((rco) => {
+    //   return rcos_detail[rco]
+    // })
+    dispatch({
+      type: "SET_DIMENSIONS",
+      property,
+      dimensions: newMultiSelect,
+    });
+  }
+
+  if (property === "neighborhood" || property === "rco_info") {
```

Comment:
Instead of hardcoding the property types here, it may be more flexible to add a new `type` field of either `multiSelect` or `buttonGroup` or something to the `filter` object, and switch off of that instead. Also, given the duplication of the tooltip and header code on both of these, might be worth DRYing that up and creating new filter components to pass into

Reply:
Makes sense.  I'll work on revising those sections.